solr/core/src/java/org/apache/solr/core/XmlConfigFile.java

```diff
@@ -125,59 +82,30 @@ public XmlConfigFile(
    */
   public XmlConfigFile(
       SolrResourceLoader loader,
-      Function<String, InputStream> fileSupplier,
       String name,
       InputSource is,
       String prefix,
       Properties substituteProps)
       throws IOException {
-    if (null == loader) throw new NullPointerException("loader");
+    Objects.requireNonNull(loader);
     this.loader = loader;
```

Comment:
The cool thing about Objects.requireNonNull is that it *returns* it's input and thus you can combine it into an existing line of code to populate a field.